### PR TITLE
Dev-4551 Vertical Line Potential End Contracts

### PR DESCRIPTION
**Pull request by jonhill13**
**High level description:**

Implements Potential End Vertical Line for Contracts.

**Technical details:**

updates to y domain.

**JIRA Ticket:**
[DEV-4551(https://federal-spending-transparency.atlassian.net/browse/DEV-4551)

**Mockup:**
https://bahdigital.invisionapp.com/share/3KIABNW4PGR#/screens/296011639

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension) Verified in dev and local that all differences are color contrast with award CONT_AWD_36C25218N2579_3600_36C25218D0071_3600. @AGuyNamedMarco 
- [x] Added Unit Tests for methods in Container Components, reducers, and helper functions
- [x] All componentWillReceiveProps, componentWillMount, and componentWillUpdate in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3339](https://federal-spending-transparency.atlassian.net/browse/DEV-3339)
- [N/A] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [N/A] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [x] Design review complete `if applicable`
- [N/A] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [x] Code review complete

- [ ] [Current End Line](https://github.com/fedspendingtransparency/usaspending-website/pull/1558) has been merged.

### CODE REVIEW

File: src/js/components/sharedComponents/VerticalLine.jsx [hunk 1]
```diff
@@ -9,7 +9,10 @@ import { throttle } from 'lodash';
 
 const propTypes = {
     xScale: PropTypes.func, // function to set line position
-    text: PropTypes.string, // text to display with line
+    text: PropTypes.oneOfType([
```
Comment (MaxwellKendall) [inline src/js/components/sharedComponents/VerticalLine.jsx]: Nice.

File: src/js/containers/award/shared/ContractGrantActivityContainer.jsx [hunk 2]
```diff
@@ -187,6 +187,7 @@ const ContractGrantActivityContainer = ({ awardId, awardType, dates }) => {
             !error.error
             && !loading
             && transactions.length > 0
+            && !badDates
```
Comment (MaxwellKendall) [inline src/js/containers/award/shared/ContractGrantActivityContainer.jsx]: What if the potential end and current end dates are the same?

File: src/js/components/award/shared/activity/ContractGrantsActivityChart.jsx [hunk 3]
```diff
@@ -290,29 +298,62 @@ const ContractGrantsActivityChart = ({
...
+                    text={endLineText}
+                    xMax={xDomain[1]}
+                    xMin={xDomain[0]}
+                    xValue={dateEndLine}
+                    showTextPosition="left"
+                    adjustmentX={padding.left}
+                    textClassname={`vertical-line__text ${endLineClassName}`}
+                    lineClassname={`vertical-line ${endLineClassName}`} />}
+                {/* potential end line */}
+                {xScale && datePotentialEndLine && <VerticalLine
```
Comment (MaxwellKendall) [inline src/js/components/award/shared/activity/ContractGrantsActivityChart.jsx]: Recommenmd introducing case where `datePotentialEndLine` and `dateEndLine` are the same!

**Comment by MaxwellKendall** [comment]
Also, @jonhill13, lets just merge this one since it has all the commits.

File: src/js/containers/award/shared/ContractGrantActivityContainer.jsx (hunk 2 above)
Comment (jonhill13) [reply src/js/containers/award/shared/ContractGrantActivityContainer.jsx]: Yeah, that is an edge case we have in Dev-4553

File: src/js/components/award/shared/activity/ContractGrantsActivityChart.jsx (hunk 3 above)
Comment (jonhill13) [reply src/js/components/award/shared/activity/ContractGrantsActivityChart.jsx]: Yeah, that is an edge case we have in Dev-4553

**Pull request merged**